.evergreen/config_generator/components/loadbalanced.py

```diff
@@ -0,0 +1,106 @@
+                command_type=EvgCommandType.SETUP,
+                file='lb-expansion.yml',
+            )
+        ]
+    }
+
+
+def make_test_task(auth: bool, ssl: bool, server_version: str):
+    auth_str = "auth" if auth else "noauth"
+    ssl_str = "ssl" if ssl else "nossl"
+    return Task(
+        name=f"loadbalanced-test-{auth_str}-{ssl_str}-{server_version}",
+        depends_on=[EvgTaskDependency(name="loadbalanced-compile")],
+        # Use `rhel8.7` distro. `rhel8.7` distro includes necessary dependency: `haproxy`.
+        run_on="rhel8.7-small",
+        tags=['loadbalanced'],
+        commands=[
+            FetchBuild.call(build_name='loadbalanced-compile'),
+            FetchDET.call(),
+            BootstrapMongoOrchestration().call(vars={
+                'AUTH': auth_str,
+                'SSL': ssl_str,
+                'MONGODB_VERSION': server_version,
+                'TOPOLOGY': 'sharded_cluster',
+                'LOAD_BALANCER': 'on',
+            }),
+            RunSimpleHTTPServer.call(),
+            FunctionCall(func='start load balancer', vars={
+                'MONGODB_URI': 'mongodb://localhost:27017,localhost:27018'
+            }),
+            RunTests().call(vars={
+                'AUTH': auth_str,
+                'SSL': ssl_str,
+                'LOADBALANCED': 'loadbalanced',
+                'CC': 'gcc',
+            })
+        ],
+    )
+
+
+def tasks():
+    yield Task(
+        name="loadbalanced-compile",
+        run_on="rhel8.7-large",
+        tags=['loadbalanced'],
+        commands=[
+            bash_exec(
+                command_type=EvgCommandType.TEST,
+                env={
+                    'CC': 'gcc',
+                    'CFLAGS': '-fno-omit-frame-pointer',
+                    'EXTRA_CONFIGURE_FLAGS': '-DENABLE_EXTRA_ALIGNMENT=OFF',
+                    'SSL': 'OPENSSL'
+                },
+                working_dir='mongoc',
+                script='.evergreen/scripts/compile.sh',
+            ),
+            UploadBuild.call()
+        ],
+    )
+
+    # Satisfy requirements specified in
+    # https://github.com/mongodb/specifications/blob/14916f76fd92b2686d8e3d1f0e4c2d2ef88ca5a7/source/load-balancers/tests/README.rst#testing-requirements
+    #
+    # > For each server version that supports load balanced clusters, drivers
+    # > MUST add two Evergreen tasks: one with a sharded cluster with both
+    # > authentication and TLS enabled and one with a sharded cluster with
+    # > authentication and TLS disabled.
+    server_versions = ['5.0', '6.0', '7.0', 'latest']
+    for server_version in server_versions:
+        yield make_test_task(auth=False, ssl=False, server_version=server_version)
+        yield make_test_task(auth=True, ssl=True, server_version=server_version)
+
+
+def variants():
+    return [
+        BuildVariant(
+            name="loadbalanced",
+            display_name="Load Balanced Tests",
+            tasks=[EvgTaskRef(name='.loadbalanced')]),
```

Comment:
```suggestion
        BuildVariant(
            name="loadbalanced",
            display_name="loadbalanced",
            tasks=[EvgTaskRef(name='.loadbalanced')]
        ),
```
Recommend keeping `display_name` equivalent to `name` for EVG CLI friendliness/consistency (no discrepancy between displayed name and actual name of the variant).